pandas/core/groupby/grouper.py

```diff
@@ -520,21 +520,21 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     any_arraylike = any(isinstance(g, (list, tuple, Series, Index, np.ndarray))
                         for g in keys)
 
-    try:
-        if isinstance(obj, DataFrame):
-            all_in_columns_index = all(g in obj.columns or g in obj.index.names
-                                       for g in keys)
-        elif isinstance(obj, Series):
-            all_in_columns_index = all(g in obj.index.names for g in keys)
-        else:
+    if (not any_callable and not any_arraylike and not any_groupers and
+            match_axis_length and level is None):
+        try:
+            if isinstance(obj, DataFrame):
+                all_in_columns_index = all(g in obj.columns or g in
+                                           obj.index.names for g in keys)
+            elif isinstance(obj, Series):
+                all_in_columns_index = all(g in obj.index.names for g in keys)
+            else:
```

Comment:
you can probably just set
``all_in_columns_index=False`` before line 523, then remove this else (or is it already being set)?

Reply:
`all_in_columns_index` is only used in that `if` right below it.

This function expects an `NDFrame`, so I think anything coming into it is going to be a `Series` or `DataFrame` (`Panel` doesn't work with this function for other reasons and is going away anyways).

I can set it to `False` if we want to be safe, but removing the `else` and that `except` block doesn't fail any tests, so it's up to you.  I personally couldn't get it to fail with anything valid I threw at `_getgrouper`

Reply:
if you can remove the try/except (and else) then do it